trading_cost.py

```diff
@@ -0,0 +1,64 @@
+"""
+A file to build cost objects.
+"""
+
+from collections import defaultdict
+
+class CurrencyCost:
+    """
+    Cost function of currency trading.
+    """
+    def __init__(self, base_ins, instruments):
+        self.base_ins = base_ins
+        self.instruments = instruments
+        self.quantity = defaultdict(int)
+        self.value = defaultdict(int)
+        self.value[base_ins] = 1
+        self.quantity[base_ins] = 1
+
+    def __call__(self, portfolio, price, liquidation):
+        if liquidation:
+            for instrument in self.instruments:
+                portfolio[-1][instrument] = (
+                    0 if instrument is not self.base_ins else 1)
+        return self.calculate(portfolio[-1], price)
+
+    def calculate(self, today, price):
+        """
+        Calculate pnl and produce cost.
+
+        We assume that our investing money is maintained in 1.
+        That make it more general to evaluate the perfoormance for the alpha.
+        For detailed, please read our wiki pages.
+        """
+        pnl = {
+            instrument : 0 for instrument in self.instruments}
+        cost = {
+            instrument : 0 for instrument in self.instruments}
+        total_value = 0
+        for instrument in today:
+            if instrument not in self.instruments:
+                raise KeyError("%s is not tradable." % instrument)
+            if (price[instrument][-1].sell is not None and
+                    price[instrument][-1].sell is not None):
```

Comment:
Hmm... I remembered that I used all ( ) for this judgment...  Fix latter.

Reply:
Well, notice that `all` will test bool(value) for all values in the given argument.
That is, the statement will not hold when `price[instrument][-1].sell` is 0.
However, it is possible that the price is exactly 0.
So I don't think `all` is feasible here.

Reply:
By the way, aren't the two conditions here equal? Why should we check `price[instrument][-1].sell is not None` twice?